packages/cli/src/lib/push/addCommon.ts

```diff
@@ -0,0 +1,85 @@
+import {CliUx} from '@oclif/core';
+import {red, dim, green} from 'chalk';
+import {BuiltInTransformers, errors} from '@coveo/push-api-client';
+import {normalizeInvalidFields} from '../flags/sourceCommonFlags';
+import {pluralizeIfNeeded} from '../utils/string';
+import dedent from 'ts-dedent';
+
+const allowedCharRegExp = new RegExp('^[a-z]+[a-z0-9_]*$');
+
+export const handleAddError = (err: unknown) => {
+  const {UnsupportedFieldError} = errors;
+  if (err instanceof UnsupportedFieldError) {
+    CliUx.ux.error('Invalid field name detected while parsing your data.', {
+      exit: false,
+    });
+    const normalizations = err.unsupportedFields.map((f) => {
+      const original = f[0];
+      const normalized = BuiltInTransformers.toLowerCase(f[1]);
+      return {
+        original,
+        normalized,
+        valid: isFieldNameValid(normalized),
+      };
+    });
+
+    const fixable = normalizations
+      .filter(({valid}) => valid)
+      .map((f) => {
```

Comment:
You can use a flatMap to avoid iterating twice
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap#for_adding_and_removing_items_during_a_map